### DISCUSSION TITLE
Group field `statePath()` conflicts with other fields displaying the same data

### DISCUSSION THREAD

**Discussion by bakerkretzmar** [filamentphp/filament#6798] [Help] [answered]
This feels like a bug to me but I'm not familiar enough with how Filament works to know for sure.

I have a resource that's kind of like a 'container' for various different kinds of form submissions. Each type of submission has its own set of fields, but in the database all the data from those fields is stored in a `form_data` JSON column.

On the create page, I've grouped the dynamic fields together so they all get saved in the nested `form_data` column automatically. Now, on the 'view' page, I'm trying to display _both_ the dynamic form fields themselves _and_ a JSON view of all the data:

```php
Group::make()
    ->statePath('form_data')
    ->schema(fn (Closure $get) => [/* compute and return dynamic list of fields specific to this form type */]),

CodeField::make('form_data')
    ->afterStateHydrated(fn ($component, $record) => $component->state(
        json_encode($record->form_data, JSON_PRETTY_PRINT)
    )),
```

This doesn't work—if I use both `->statePath('form_data')` and `CodeField::make('form_data')`, neither of them display correctly. The group with dynamic fields shows them all as empty, and the code field breaks and doesn't show anything (replacing it with a text field shows that its value is `[object Object]`. Both the group and the code field work alone if the other one isn't on the page.

I can work around this by just calling the code field `_form_data`, which doesn't exist, but that feels gross.

**Comment by danharrin** [comment]
Yeah it is not a bug, it is just how the package works. The state path and the name of the fields basically tell the package where to store data in the form's data array, and if both fields are trying to write to the same place then its not gonna work.

**Reply by bakerkretzmar** [reply]
@danharrin that makes sense, but I'm not trying to write there, just read. I should have been more specific in my question—those fields are only _both_ shown on the `view` page:

```php
Group::make()
    ->statePath('form_data')
    ->schema(fn (Closure $get) => [/* compute and return dynamic list of fields specific to this form type */]),

CodeField::make('form_data')
    ->afterStateHydrated(fn ($component, $record) => $component->state(
        json_encode($record->form_data, JSON_PRETTY_PRINT)
    ))
    ->visibleOn('view'),
```

**Reply by danharrin** [reply]
Is CodeField custom or from a plugin?

**Reply by bakerkretzmar** [reply]
It's from a plugin but using one of the built-in fields does the same thing.

This doesn't affect the Group schema (the individual fields are filled correctly) but its content is `[object Object]`:

```php
TextInput::make('form_data')
```

This makes the Group schema not get filled at all (all the fields are empty) _and_ it still just shows `[object Object]`:

```php
TextInput::make('form_data')
    ->afterStateHydrated(fn ($component, $record) => $component->state(json_encode($record?->form_data))),
```

**Reply by danharrin** [reply, marked as answer]
The problem is that while a field is read-only, it still has its own data. For a truly stateless "field", you just want a layout component, which is able to get the state of another field, using `$get`. The easiest way to implement would be `View::make('code')` and then inside the `code.blade.php` `{{ json_encode($evaluate(fn ($get) => $get('form_data'))) }}`

**Reply by bakerkretzmar** [reply]
Hmm, `Placeholder` works fine though. Calling `->dehydrated(false)` on the `TextInput` and `CodeField` doesn't though. Would this have something to do with `Placeholder` returning `false` from `shouldEvaluateWithState`?

**Reply by bakerkretzmar** [reply]
Didn't see your comment, thanks. That makes sense but if each field has its _own_ state I don't understand why mutating/formatting one field would cause the others to behave weirdly. Or does mutating that state actually mutate it globally, not just within the context of the field?

**Reply by bakerkretzmar** [reply]
Okay I think I answered my own question there—it looks like `CodeField` (and I assume lots of built-in ones) 'entangle' their state, which means they do use and mutate form state from the `$data` property. Which makes sense I guess. I'll keep faking it with `_form_data` so I don't have to rebuild the field, but you're right a simple view would be easier too. Thanks!

**Reply by danharrin** [reply]
>  I don't understand why mutating/formatting one field would cause the others to behave weirdly

The state is stored in a property in the Livewire component. When you mutate it, it mutates it on the property. All fields with the same name read from the same property.
